File: scripts/reference-solutions/uniform-cohort-batches.cpp
#include <algorithm>
#include <iostream>
#include <vector>
using namespace std;

int main() {
  ios::sync_with_stdio(false);
  cin.tie(nullptr);

  long long n, d, c;
  cin >> n >> d >> c;
  vector<long long> skills(n);
  for (long long& value : skills) {
    cin >> value;
  }
  sort(skills.begin(), skills.end());

  long long batches = 0;
  long long i = 0;
  while (i < n) {
    ++batches;
    long long start = skills[i];
    long long size = 0;
    while (i < n && skills[i] - start <= d && size < c) {
      ++i;
      ++size;
    }
  }

  cout << batches << '\n';
  return 0;
}

File: scripts/reference-solutions/uniform-cohort-batches.py
import sys


def main():
    data = sys.stdin.buffer.read().split()
    n = int(data[0])
    d = int(data[1])
    c = int(data[2])
    skills = list(map(int, data[3 : 3 + n]))
    skills.sort()

    batches = 0
    i = 0
    while i < n:
        batches += 1
        start = skills[i]
        size = 0
        while i < n and skills[i] - start <= d and size < c:
            i += 1
            size += 1
    print(batches)


main()
